### PR TITLE
TIP-721: Make API work

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

This PR fixes some failing integration tests of the API.

There is still failing ones:
- every tests that require the PQB, we'll see that again once the PQB is fully functionnal,
- some tests that use the API ProductRepository => it cannot work as it uses Doctrine QB, which is not present anymore in the PQB. This part needs to be discussed and so will be fixed later.

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/CreateProductIntegration.php [hunk 1]
```diff
@@ -881,7 +881,7 @@ public function testResponseWhenProductAlreadyExists()
             'errors'  => [
                 [
                     'property' => 'identifier',
-                    'message'  => 'The value simple is already set on another product for the unique attribute sku',
+                    'message'  => 'The same identifier is already set on another product',
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/CreateProductIntegration.php:884]: check with @skeleton if it's OK

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteProductIntegration.php [hunk 2]
```diff
@@ -23,16 +23,15 @@ public function testDeleteAProduct()
         $client = $this->createAuthenticatedClient();
 
         $this->assertCount(3, $this->get('pim_catalog.repository.product')->findAll());
-        // TODO: TIP-613: This repository does not exist anymore. We need to find another way to count.
-        $this->assertEquals(30, $this->get('pim_catalog.repository.product_value_counter')->count());
+        $this->assertProductRawValuesEquals(27);
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteProductIntegration.php]: imho we don't even need to test that
I don't understand why it's relevant for the API

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteProductIntegration.php [hunk 3]
```diff
@@ -23,16 +23,15 @@ public function testDeleteAProduct()
...
 
         $client->request('DELETE', 'api/rest/v1/products/foo');
 
         $response = $client->getResponse();
         $this->assertSame(Response::HTTP_NO_CONTENT, $response->getStatusCode());
 
         $this->assertCount(2, $this->get('pim_catalog.repository.product')->findAll());
-        $this->assertEquals(2, $this->get('pim_catalog.repository.product_value_counter')->count());
+        $this->assertProductRawValuesEquals(0);
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteProductIntegration.php]: same here

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/PartialUpdateProductIntegration.php [hunk 4]
```diff
@@ -1285,7 +1285,7 @@ public function testResponseWhenProductAlreadyExists()
             'errors'  => [
                 [
                     'property' => 'identifier',
-                    'message'  => 'The value product_family is already set on another product for the unique attribute sku',
+                    'message'  => 'The same identifier is already set on another product',
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/PartialUpdateProductIntegration.php:1288]: same, check with @skeleton

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/CreateProductIntegration.php (hunk 1 above)
Comment (skeleton) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/CreateProductIntegration.php:884]: it's ok, we accept to change message on the API (even if we should avoid, but in this case, the message is cleaner like that)

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteProductIntegration.php (hunk 2 above)
Comment (skeleton) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteProductIntegration.php]: not really relevant for the API, bu we wanted to check if product values are really deleted when you delete a product. should be done on a low level test (on Remover for instance)

Comment (jjanvier) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteProductIntegration.php]: ok, so we can drop it then
as values are totally part of the product now, the tests are no longer useful

**Review by jjanvier — APPROVED** [review]
GTM once fixed :+1: 

thanks :)

**Review by skeleton — APPROVED** [review]
Thanks a lot to fix the API :)
GTM

**Pull request merged**